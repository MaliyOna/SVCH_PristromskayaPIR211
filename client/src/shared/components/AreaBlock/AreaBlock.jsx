import React from 'react';
import './AreaBlock.scss';

export function AreaBlock(props) {
  function handleClick() {
    if (props.onClick) {
      props.onClick();
    }
  }

  return (
    <div className='areaBlock' onClick={handleClick}>
      <div className='areaBlock__area'>{props.area}</div>
      <div className='areaBlock__brigade'>{props.brigade}</div>
    </div>
  );
}