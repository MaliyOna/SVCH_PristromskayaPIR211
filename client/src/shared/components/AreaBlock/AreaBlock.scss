.areaBlock {
  box-shadow: 0 .2vw 1vw 0 rgba(0,0,0,0.16);
  height: 3vw;
  margin-bottom: 1vw;
  display: flex;
  flex-direction: row;
  font-size: 1vw;
}

.areaBlock__area, .areaBlock__brigade {
  margin-top: auto;
  margin-bottom: auto;
}

.areaBlock__area {
  margin-left: .5vw;
  margin-right: auto;
}

.areaBlock__brigade {
  margin-right: .5vw;
  margin-left: auto;
}
