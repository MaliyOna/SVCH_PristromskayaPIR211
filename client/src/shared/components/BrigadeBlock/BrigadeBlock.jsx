import React from 'react';
import './BrigadeBlock.scss';

export function BrigadeBlock(props) {
  return (
    <div className='brigadeBlock'>
      <div className='brigadeBlock__brigade'>{props.brigade}</div>
    </div>
  );
}