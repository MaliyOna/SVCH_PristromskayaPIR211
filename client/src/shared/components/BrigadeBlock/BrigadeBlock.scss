.brigadeBlock {
  box-shadow: 0 .2vw 1vw 0 rgba(0,0,0,0.16);
  height: 3vw;
  margin-bottom: 1vw;
  font-size: 1vw;
  display: flex;
}

.brigadeBlock__brigade {
  margin-top: auto;
  margin-bottom: auto;
  margin-left: .5vw;
}