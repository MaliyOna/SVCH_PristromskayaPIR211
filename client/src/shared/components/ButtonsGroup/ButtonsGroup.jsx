import * as React from 'react';
import Button from '@mui/material/Button';
import ButtonGroup from '@mui/material/ButtonGroup';
import { GroupButton } from '../GroupButton/GroupButton';

export function ButtonsGroup(props) {
  return (
    <ButtonGroup variant="contained">
      <GroupButton text="Сохранить" function={() => props.save()}/>
      <GroupButton text="Отмена" function={() => props.close()}/>
      <GroupButton text="Удалить" function={() => props.delete()}/>
    </ButtonGroup>
  );
}