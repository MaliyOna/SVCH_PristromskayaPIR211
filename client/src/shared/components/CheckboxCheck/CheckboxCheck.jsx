import React, { useEffect } from 'react';
import './CheckboxCheck.scss';
import Checkbox from '@mui/material/Checkbox';
import FormControlLabel from '@mui/material/FormControlLabel';

export function CheckboxCheck(props) {
    const [checked, setChecked] = React.useState(props.admin);

    useEffect(() => {
        setChecked(props.admin);
    }, [props.admin]);

    function handleChange(event) {
        setChecked(checked ? false : true);
        props.onChange(checked ? false : true);
    };

    return (
        <div className='checkboxCheck'>
            <FormControlLabel
                control={
                    <Checkbox
                        checked={checked}
                        onChange={handleChange}
                    />
                }
                label="Права админа"
            />

        </div>
    )
}