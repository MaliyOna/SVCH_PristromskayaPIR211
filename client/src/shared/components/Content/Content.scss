.content {
    font-size: 1vw;
    padding: .5vw;
    margin: 1vw 6vw;
    min-height: 20vw;
    border: .2vw solid #1976d2;
}