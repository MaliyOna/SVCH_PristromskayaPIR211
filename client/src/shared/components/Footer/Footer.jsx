import React from 'react';
import './Footer.scss';

export function Footer() {
  return (
    <div className='footer'>
      <div className='footer__information'>
        Данный веб-сайт разработан для управления различными бригадами для уборки, а также указывает где и какие бриды должны убирать. Также видно время работы бригады.
      </div>
      
      <div className='footer__contacts'>
        <div>Email: uborka@gmail.com</div>
        <div>Телефон: +375 22 222 22 22</div>
      </div>
    </div>
  );
}