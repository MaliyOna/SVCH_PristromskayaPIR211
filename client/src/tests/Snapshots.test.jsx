import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/react';

import { Footer } from '../shared/components/Footer/Footer';
import { AreaBlock } from '../shared/components/AreaBlock/AreaBlock';
import { BrigadeBlock } from '../shared/components/BrigadeBlock/BrigadeBlock';
import { ScheduleBlock } from '../shared/components/ScheduleBlock/ScheduleBlock';
import { CheckboxCheck } from '../shared/components/CheckboxCheck/CheckboxCheck';
import { Input } from '../shared/components/Input/Input';
import { ButtonsGroup } from '../shared/components/ButtonsGroup/ButtonsGroup';
import { Content } from '../shared/components/Content/Content';
import { PopupWindow } from '../shared/components/PopupWindow/PopupWindow';
import { SimpleAvatar } from '../shared/components/SimpleAvatar/SimpleAvatar';

test('renders Footer component correctly', () => {
  const { container } = render(<Footer />);
  expect(container).toMatchSnapshot();
});

test('renders AreaBlock component correctly', () => {
  const { container } = render(<AreaBlock />);
  expect(container).toMatchSnapshot();
});

test('renders BrigadeBlock component correctly', () => {
  const { container } = render(<BrigadeBlock />);
  expect(container).toMatchSnapshot();
});

test('renders ScheduleBlock component correctly', () => {
  const { container } = render(<ScheduleBlock />);
  expect(container).toMatchSnapshot();
});

test('renders CheckboxCheck component correctly', () => {
  const { container } = render(<CheckboxCheck />);
  expect(container).toMatchSnapshot();
});

test('renders Input component correctly', () => {
  const { container } = render(<Input />);
  expect(container).toMatchSnapshot();
});

test('renders ButtonsGroup component correctly', () => {
  const { container } = render(<ButtonsGroup />);
  expect(container).toMatchSnapshot();
});

test('renders Content component correctly', () => {
  const { container } = render(<Content />);
  expect(container).toMatchSnapshot();
});

test('renders PopupWindow component correctly', () => {
  const { container } = render(<PopupWindow />);
  expect(container).toMatchSnapshot();
});

test('renders SimpleAvatar component correctly', () => {
  const { container } = render(<SimpleAvatar />);
  expect(container).toMatchSnapshot();
});
