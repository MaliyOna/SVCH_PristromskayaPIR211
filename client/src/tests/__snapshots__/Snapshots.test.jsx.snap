// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AreaBlock component correctly 1`] = `
<div>
  <div
    class="areaBlock"
  >
    <div
      class="areaBlock__area"
    />
    <div
      class="areaBlock__brigade"
    />
  </div>
</div>
`;

exports[`renders AreaBlock component correctly 2`] = `
<div>
  <div
    class="areaBlock"
  >
    <div
      class="areaBlock__area"
    />
    <div
      class="areaBlock__brigade"
    />
  </div>
</div>
`;

exports[`renders AreaBlock component correctly 3`] = `
<div>
  <div
    class="brigadeBlock"
  >
    <div
      class="brigadeBlock__brigade"
    />
  </div>
</div>
`;

exports[`renders BrigadeBlock component correctly 1`] = `
<div>
  <div
    class="brigadeBlock"
  >
    <div
      class="brigadeBlock__brigade"
    />
  </div>
</div>
`;

exports[`renders ButtonsGroup component correctly 1`] = `
<div>
  <div
    class="MuiButtonGroup-root MuiButtonGroup-contained css-1d9bq45-MuiButtonGroup-root"
    role="group"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-firstButton css-sghohy-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Сохранить
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-middleButton css-sghohy-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Отмена
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-lastButton css-sghohy-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Удалить
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`renders CheckboxCheck component correctly 1`] = `
<div>
  <div
    class="checkboxCheck"
  >
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
      >
        <input
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Права админа
      </span>
    </label>
  </div>
</div>
`;

exports[`renders Content component correctly 1`] = `
<div>
  <div
    class="content"
  />
</div>
`;

exports[`renders Footer component correctly 1`] = `
<div>
  <div
    class="footer"
  >
    <div
      class="footer__information"
    >
      Данный веб-сайт разработан для управления различными бригадами для уборки, а также указывает где и какие бриды должны убирать. Также видно время работы бригады.
    </div>
    <div
      class="footer__contacts"
    >
      <div>
        Email: uborka@gmail.com
      </div>
      <div>
        Телефон: +375 22 222 22 22
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Input component correctly 1`] = `
<div>
  <div
    class="input"
  >
    <div
      class="input__title"
    />
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-bz5ng3-MuiInputBase-root-MuiInput-root"
    >
      <input
        class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`renders PopupWindow component correctly 1`] = `<div />`;

exports[`renders ScheduleBlock component correctly 1`] = `
<div>
  <div
    class="scheduleBlock"
  >
    <div
      class="scheduleBlock__brigade"
    />
    <div
      class="scheduleBlock__schedule"
    />
  </div>
</div>
`;

exports[`renders SimpleAvatar component correctly 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
    >
      <img
        alt="Remy Sharp"
        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="/images.jpg"
      />
    </div>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
